docs/testplan/stormond-test-plan.md

```diff
@@ -0,0 +1,274 @@
+2024 May 28 17:50:55.695842 sonic-device INFO pmon#stormond[96]: /usr/share/stormond/fsio-rw-stats.json not present.
+2024 May 28 17:50:55.696967 sonic-device INFO pmon#stormond[96]: Starting Storage Monitoring Daemon
+2024 May 28 17:50:55.697958 sonic-device INFO pmon#stormond[96]: Storage Device: sda, Device Model: InnoDisk Corp. - mSATA 3IE3, Serial: BCA12310040200027
+2024 May 28 17:50:55.701215 sonic-device INFO pmon#stormond[96]: Polling Interval set to 3600 seconds
+2024 May 28 17:50:55.702037 sonic-device INFO pmon#stormond[96]: FSIO JSON file Interval set to 86400 seconds
+2024 May 28 17:50:56.068452 sonic-device INFO pmon#stormond[96]: Storage Device: sda, Firmware: S16425cG, health: 99%, Temp: 30C, FS IO Reads: 36708, FS IO Writes: 68792
+2024 May 28 17:50:56.068602 sonic-device INFO pmon#stormond[96]: Latest FSIO Reads: 13854, Latest FSIO Writes: 6386
+2024 May 28 17:50:56.068671 sonic-device INFO pmon#stormond[96]: Disk IO Reads: 26092, Disk IO Writes: 16047, Reserved Blocks: 63
+2024 May 28 17:51:04.821947 sonic-device INFO pmon#supervisord 2024-05-28 17:51:04,820 INFO success: stormond entered RUNNING state, process has stayed up for > than 10 seconds (startsecs)
+``` 
+
+#### Test Case 4.1.1: Initialize stormond and verify the following:
+  1. Stormond is spawned with a valid process ID.
+  2. The daemon starts without errors and logs the start-up message.
+  3. The polling interval and FSIO JSON file interval are set correctly.
+  4. `STATE_DB` Status: Empty
+  5. JSON File: Empty
+  6. PROCFS Status: Initial Values
+  7. JSON File Synced With `STATE_DB`?: Yes
+  8. `stormond` Restarted: Yes
+
+#### Test Case 4.1.2: Verify the running state of stormond:
+  1. The daemon logs the storage device attributes: health, temperature, FS IO Reads/Writes, Disk IO Reads/Writes and Reserved Blocks to `STATE_DB`.
+  2. Ensure the latest FSIO Reads and Writes are updated correctly.
+  3. `STATE_DB` Status: Initial values
+  4. JSON File: Empty
+  5. PROCFS Status: Initial Values
+  6. JSON File Synced With `STATE_DB`?: No
+
+### 4.2 Bind Mount Operations
+
+**Objective**: Confirm that the bind mount operations work correctly for the daemon’s files.
+
+**Test Case 4.2.1**: Verify bind mount in the container and host:
+  1. Ensure that the `/host/pmon/stormond/` directory on the host reflects the correct bind mount from the container.
+  2. Check that the `/usr/share/stormond/` directory on the container contains the expected file(s).
+  3. Verify the file permissions, ownership and inode number are identical in the bind-mounted directories in container and host.
+
+    **Container:**
+    ```
+    root@sonic-device:/# ls -il /usr/share/stormond/
+    total 4
+    527873 -rw-r--r-- 1 root root 9 May 28 17:53 bindmnt.txt
+    root@sonic-device:/#
+    ```
+
+    **Host**
+    ```
+    admin@sonic-device:~$ ls -il /host/pmon/stormond/
+    total 4
+    527873 -rw-r--r-- 1 root root 9 May 28 17:53 bindmnt.txt
+    ```
+
+### 4.3 Planned Reboots
```

Comment:
Do we need any planned power down/up sequence tests as well?

Reply:
I don't think that is needed, because planned power cycles are identical to cold reboots from the perspective of the daemon and system attributes that are measured by the daemon.